cosmos/x/evm/plugins/state/genesis.go

```diff
@@ -22,24 +22,34 @@ package state
 
 import (
 	"math/big"
+	"sort"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"github.com/ethereum/go-ethereum/core"
-
 	"pkg.berachain.dev/polaris/eth/common"
+	"pkg.berachain.dev/polaris/eth/core"
 )
 
 // InitGenesis takes in a pointer to a genesis state object and populates the KV store.
 func (p *plugin) InitGenesis(ctx sdk.Context, ethGen *core.Genesis) {
 	p.Reset(ctx)
 
-	// Iterate over the genesis accounts and set the balances.
-	for address, account := range ethGen.Alloc {
+	// Sort the addresses so that they are in a consistent order.
+	sortedAddresses := make(core.SortableAddresses, 0, len(ethGen.Alloc))
+	for address := range ethGen.Alloc {
+		sortedAddresses = append(sortedAddresses, address)
+	}
+	sort.Sort(sortedAddresses)
+
+	// Iterate over the sorted genesis accounts and set the balances.
+	for _, address := range sortedAddresses {
 		// TODO: technically wrong since its overriding / hacking the auth keeper and
 		// we are using the nonce from the account keeper as well.
+		account := ethGen.Alloc[address]
 		p.CreateAccount(address)
-		p.SetBalance(address, account.Balance)
+		if account.Balance != nil {
+			p.SetBalance(address, account.Balance)
+		}
 		if account.Code != nil {
 			p.SetCode(address, account.Code)
```

Comment:
add check to make sure if code isn't nil nonce >= 1 if the proper hard fork is enable (i forget the exact one)

Reply:
also add a check that nonce == Statedb.GetNonce() and then add note that shows that because account keeper is earlier in InitGenesisOrder, the noncing here is redundant

Reply:
its eip 161 which was released with eip155 (theres no check for 161 specifically)

Reply:
check `EIP155Block`

Reply:
i think geth has a VerifyGenesis function

Reply:
well by default, its 2million smth but it can be modified in the given genesis to be 0, in which case the nonce must be >= 1 for an account with code

Reply:
actually the geth default is 2mil smth and polaris default is 0

Reply:
yeah cause its using the one from actual live mainnet chainid 1